File: tests/expected/cast.kt

fun cast_types() {
    val a = 1.toDouble().toInt()
    println("$a")
}

fun cast_ctypes() {
    val a = 1.toShort()
    val b = a
    println("$b")
    val c = 1.toLong()
    val d = c
    println("$d")
    val e = 1.toULong()
    val f = e
    println("$f")
}

fun main(argv: Array<String>) {
    cast_types()
    cast_ctypes()
}

File: tests/expected/cast.dart
// @dart=2.9
import 'package:sprintf/sprintf.dart';

cast_types() {
  final int a = 1.toDouble().toInt();
  print(sprintf("%s", [a]));
}

cast_ctypes() {
  final int a = 1.toInt();
  final b = a;
  print(sprintf("%s", [b]));
  final int c = 1.toInt();
  final d = c;
  print(sprintf("%s", [d]));
  final int e = 1.toInt();
  final f = e;
  print(sprintf("%s", [f]));
}

main(List<String> argv) {
  cast_types();
  cast_ctypes();
}
